### PR TITLE
Add EF.Functions.IsNumeric

### PR DESCRIPTION
**Pull request by RaymondHuy**
Fix: https://github.com/dotnet/efcore/issues/23114

### CODE REVIEW

File: src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            Check.NotNull(arguments, nameof(arguments));
+            Check.NotNull(logger, nameof(logger));
+
+            return _methodInfo.Equals(method)
+                ? _sqlExpressionFactory.Convert(
+                    _sqlExpressionFactory.Function(
+                        "ISNUMERIC",
+                        new[] { arguments[1] },
+                        nullable: true,
+                        argumentsPropagateNullability: new[] { true },
```
Comment (bricelam) [inline src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs:58]:
```suggestion
                        nullable: false,
                        argumentsPropagateNullability: new[] { false },
```

Comment (bricelam) [reply src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs:58]: `ISNUMERIC(NULL)` returns 0, not NULL

File: src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            MethodInfo method,
+            IReadOnlyList<SqlExpression> arguments,
+            IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+        {
+            Check.NotNull(method, nameof(method));
+            Check.NotNull(arguments, nameof(arguments));
+            Check.NotNull(logger, nameof(logger));
+
+            return _methodInfo.Equals(method)
+                ? _sqlExpressionFactory.Convert(
```
Comment (bricelam) [inline src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs:53]: Why `Convert`? AFAICT, it's not needed.

**Review by bricelam — CHANGES_REQUESTED** [review]

File: src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs (hunk 2 above)
Comment (RaymondHuy) [reply src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs]: Thanks @bricelam  I've just updated my PR.

**Review by bricelam — APPROVED** [review]

File: src/EFCore.SqlServer/Query/Internal/SqlServerMethodCallTranslatorProvider.cs [hunk 3]
```diff
@@ -41,7 +41,8 @@ public SqlServerMethodCallTranslatorProvider([NotNull] RelationalMethodCallTrans
                     new SqlServerMathTranslator(sqlExpressionFactory),
                     new SqlServerNewGuidTranslator(sqlExpressionFactory),
                     new SqlServerObjectToStringTranslator(sqlExpressionFactory),
-                    new SqlServerStringMethodTranslator(sqlExpressionFactory)
+                    new SqlServerStringMethodTranslator(sqlExpressionFactory),
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Query/Internal/SqlServerMethodCallTranslatorProvider.cs:44]: A-B-C

File: src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public class SqlServerIsNumericFunctionTranslator : IMethodCallTranslator
+    {
+        private readonly ISqlExpressionFactory _sqlExpressionFactory;
+
+        private static readonly MethodInfo _methodInfo = typeof(SqlServerDbFunctionsExtensions)
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs:26]: Was just wondering why this did not error out for nullable.

Comment (RaymondHuy) [reply src/EFCore.SqlServer/Query/Internal/SqlServerIsNumericFunctionTranslator.cs:27]: Thanks @smitpatel  I've just updated my PR.

**Comment by RaymondHuy** [comment]
Hi @smitpatel  @bricelam  Are there anything else I need to change so that this PR can be merged ?

**Comment by smitpatel** [comment]
@RaymondHuy - I will process this PR and merge it to master in a day or two. I don't believe we need anymore changes here. Thank for you contribution.

**Pull request merged**